tests/php/core-api/wpcom-endpoints/test-class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -396,4 +409,60 @@ public function menu_item_url_data() {
 			),
 		);
 	}
+
+	/**
+	 * Tests parsing an update count.
+	 *
+	 * @param string $menu_item Menu item.
+	 * @param string $expected  Parsed menu title & count. Or not.
+	 *
+	 * @throws \ReflectionException Noop.
+	 * @dataProvider menu_item_update_data
+	 * @covers ::parse_update_count
+	 */
+	public function test_parse_update_count( $menu_item, $expected ) {
+		$class = new ReflectionClass( 'WPCOM_REST_API_V2_Endpoint_Admin_Menu' );
```

Comment:
Out of interest why do we not do more integration style tests on the endpoint? I see that Reflection is good for unit testing but does that provide the necessary confidence that the endpoint behaves as a whole when used?

Was this a conscious decision because mocking out the dependencies of the endpoint was too complex?

Reply:
I only use it to make the private method accessible for testing.